auth/user_mgt.go

```diff
@@ -468,6 +469,11 @@ func IsEmailAlreadyExists(err error) bool {
 	return hasAuthErrorCode(err, emailAlreadyExists)
 }
 
+// IsEmailNotFound checks if the given error was due to the user record corresponding to the email not being found.
```

Comment:
Lets also add a test case at https://github.com/firebase/firebase-admin-go/blob/4121c50b610f0ce408a2978cf14b306becf276d3/auth/user_mgt_test.go#L510

Reply:
I added a test to `email_action_links_test.go` as that is where all the `PasswordResetLinkWithSettings` tests are located.